enterprise/views.py

```diff
@@ -360,81 +318,14 @@ def get_course_specific_consent(self, request, course_id):
-        if customer is None:
-            # If we can't get an EnterpriseCustomer from the pipeline, then we don't really
-            # have enough state to do anything meaningful. Just send the user to the login
-            # screen; if they want to sign in with an Enterprise-linked SSO, they can do
-            # so, and the pipeline will get them back here if they need to be.
-            return redirect('signin_user')
-
-        # Quarantine the user to this module.
-        self.quarantine(request)
-
-        failure_url = request.GET.get('failure_url')
-
-        context_data = self.get_default_context(customer, platform_name)
-
-        account_specific_context = {
-            'consent_request_prompt': CONSENT_REQUEST_PROMPT.format(  # pylint: disable=no-member
-                enterprise_customer_name=customer.name
-            ),
-            'confirmation_alert_prompt': CONFIRMATION_ALERT_PROMPT.format(  # pylint: disable=no-member
-                enterprise_customer_name=customer.name
-            ),
-            'confirmation_alert_prompt_warning': CONFIRMATION_ALERT_PROMPT_WARNING.format(  # pylint: disable=no-member
-                enterprise_customer_name=customer.name,
-            ),
-            'LANGUAGE_CODE': get_language_from_request(request),
-            'platform_name': platform_name,
-            'enterprise_customer_name': customer.name,
-            "course_id": None,
-            "course_specific": False,
-            'enrollment_deferred': False,
-            'failure_url': failure_url,
-            'requested_permissions': [
-                _('your enrollment in all sponsored courses'),
-                _('your learning progress'),
-                _('course completion'),
-            ],
-            'enterprise_customer': customer,
-            'welcome_text': self.welcome_text.format(platform_name=platform_name),
-            'enterprise_welcome_text': self.enterprise_welcome_text.format(
-                enterprise_customer_name=customer.name,
-                platform_name=platform_name,
-                strong_start='<strong>',
-                strong_end='</strong>',
-            ),
-        }
-
-        context_data.update(account_specific_context)
-
-        return render(request, 'enterprise/grant_data_sharing_permissions.html', context=context_data)
-
     def get(self, request):
         """
         Render a form to collect user input about data sharing consent.
-
-        Based on a URL parameter, the form rendered will either be course-specific or appropriate
-        for granting consent for all courses sponsored by the EnterpriseCustomer.
         """
         # Verify that all necessary resources are present
         verify_edx_resources()
         course = request.GET.get('course_id')
-        if course:
-            return self.get_course_specific_consent(request, course)
-        return self.get_account_consent(request)
+        return self.get_course_specific_consent(request, course)
```

Comment:
what will happen if course is None? is it ever possible that it will be anymore?